### ISSUE TITLE
Use micrometer context-propagation to propagate auth between threadlocal and reactor ops

### ISSUE THREAD

**Issue by ttddyy** [spring-projects/spring-security#13733]
In spring mvc with reactive API, it uses a custom reactor subscriber, [SecurityReactorContextSubscriber](https://github.com/spring-projects/spring-security/blob/main/config/src/main/java/org/springframework/security/config/annotation/web/configuration/SecurityReactorContextConfiguration.java#L83), to propagate the authentication and servlet request and response to the reactor operation chain. (Defined in `SecurityReactorContextConfiguration` imported by `OAuth2ImportSelector`)

The [micrometer context-propagation](https://github.com/micrometer-metrics/context-propagation) library provides support for propagating such objects between threadlocal and reactor operation chains.

It is nice to migrate to it which aligns how micrometer observations are propagated as well as avoiding reactor corner cases that may not able to propagate by `Hooks.onLastOperator`.

cc @sjohnr, @chemicL

relates to https://github.com/spring-projects/spring-security/issues/11973

**Comment by rstoyanchev** [comment]
Currently we have a [SecurityContextThreadLocalAccessor](https://github.com/spring-projects/spring-graphql/blob/main/spring-graphql/src/main/java/org/springframework/graphql/execution/SecurityContextThreadLocalAccessor.java) in Spring for GraphQL that I can see would be more generally applicable. Not sure if that belongs in this issue or another.

**Comment by jzheaux** [comment]
@sjohnr, can you clarify here the difference between this ticket and https://github.com/spring-projects/spring-security/issues/16665?

**Comment by chemicL** [comment]
@jzheaux Hey, I think I have some context here. I believe the change in #16665 adds optional support for the context-propagation in certain conditions:
> to be used with [blocking execution](https://docs.spring.io/spring-framework/reference/web/webflux/config.html#webflux-config-blocking-execution) when [automatic context propagation](https://docs.spring.io/spring-boot/reference/actuator/observability.html#actuator.observability.context-propagation) is enabled and the [context-propagation](https://github.com/micrometer-metrics/context-propagation) library is on the classpath.

while this issue in particular is suggesting that the existing mechanism that uses `onLastOperatorHook` be replaced entirely. We discussed this with @sjohnr at one point and agreed that the security propagation can't rely on an optional dependency, hence context-propagation library would have to be required and the automatic propagation mode would need to be enabled. That is too much to ask from the users, but it can be helpful in certain scenarios that are already meeting these conditions.

If @ttddyy agrees and the optional support gives the users everything required, we could close this issue.

**Comment by ttddyy** [comment]
Yes, I agree — making it optional makes sense, and the necessary components are now in place.
With context propagation automatically enabled via the service loader, it provides a convenient experience for users.

To improve the developer experience further, it would be helpful to update existing consumers of these propagated objects.
For example, the current `SecurityReactorContextSubscriber` is used by `ServletBearerExchangeFilterFunction` and `ServletOAuth2AuthorizedClientExchangeFilterFunction`.

It would be great if they could auto-detect the presence of the new thread-local accessors and use them when available — particularly `ServletBearerExchangeFilterFunction`, since it only relies on `Authentication`.
Alternatively, introducing new implementations that support this pattern could be another approach.

If it's too much for now, it is probably nice to update the Javadoc to note that `SecurityContext` can now be retrieved from `SecurityContextHolder` when context propagation is enabled.

**Comment by philsttr** [comment]
dup of #13123 ?
